### PR TITLE
apply updates to comply with recent core USD dev branch commits

### PR DESCRIPTION
**Pull request by mattyjams**
In particular, this PR absorbs the following core USD commits:
https://github.com/PixarAnimationStudios/USD/commit/82e343c42b730a3674a766aab6be1b4f1c0ca9ef
https://github.com/PixarAnimationStudios/USD/commit/310d9fbc2786b9ec5a3278c033b16ce713b92f49
https://github.com/PixarAnimationStudios/USD/commit/b0fc236ac695b6c66de50897356209ed7c21865a
https://github.com/PixarAnimationStudios/USD/commit/eff8cb850e44aab0d6afa89c80706e88984fefe6

I updated build.md to reflect that the most recent core USD commit supported by maya-usd is the one currently on the tip of the core USD dev branch (https://github.com/PixarAnimationStudios/USD/commit/8dcbcc9b46ce8f04d3ae330772b8936bef5cc9dd).

Note that #395 is also required if you want to build the maya-usd dev branch against the core USD dev branch.

### CODE REVIEW

**Comment by kxl-adsk** [comment]
@mattyjams I think we will wait for 20.05 to be out of the door before adopting this change. Does it work for you?

**Comment by mattyjams** [comment]
@kxl-adsk: Internally, we're ok with that, as we're already merging these changes into the branch we build for deployment at Pixar. This is more for the benefit of other users/studios who might be trying to build against the dev branch of core USD. We've fielded a bunch of queries already asking about the USD dev branch having broken maya-usd, and we're having to point them at this PR and #395. So the sooner we can get this in, the better really.

File: lib/render/vp2RenderDelegate/proxyRenderDelegate.cpp [hunk 1]
```diff
@@ -537,10 +537,14 @@ bool ProxyRenderDelegate::getInstancedSelectionPath(
     // therefore drawInstID is usdInstID plus 1 considering VP2 defines the
     // instance ID of the first instance as 1.
     const int drawInstID = intersection.instanceID();
+    const int usdInstID = drawInstID - 1;
+#if defined(USD_IMAGING_API_VERSION) && USD_IMAGING_API_VERSION >= 13
```
Comment (HdC-adsk) [inline lib/render/vp2RenderDelegate/proxyRenderDelegate.cpp:541]: Is there any user-visible change using the new code?

Comment (mattyjams) [reply lib/render/vp2RenderDelegate/proxyRenderDelegate.cpp:541]: @HdC-adsk: No, there should be no change in what gets drawn in the viewport. This was just cleaning up the API around querying scene delegates for an instancer's prim path.

There's more detail on the core USD change that goes with this here:
https://github.com/PixarAnimationStudios/USD/commit/82e343c42b730a3674a766aab6be1b4f1c0ca9ef

**Comment by mattyjams** [comment]
Rebased following the merge of #388.

**Comment by mattyjams** [comment]
Updated build.md to reflect that core USD release 20.05 is supported.

**Comment by seando-adsk** [comment]
@mattyjams I have built USD 20.05 locally on Windows and needed to make some extra changes on top of yours to get everything to build. All these boost changes are because (on Windows) the version was changed from 1.65.1 to 1.70.0. I've attached a patch file of my changes here:

[USD_Patch_for_Matt.patch.txt](https://github.com/Autodesk/maya-usd/files/4483597/USD_Patch_for_Matt.patch.txt)

Would you prefer to incorporate these changes into your PR, or should I create my own PR?

Thanks, Sean

**Comment by mattyjams** [comment]
Ahh, good catch, thanks @seando-adsk! I just applied that patch to the change branch, so those changes should be present in fb2f13fc16ba820aa0cb0393073849a06c8217c4.

**Comment by seando-adsk** [comment]
@mattyjams Sorry we found a problem when building on the Mac. Could you add in this one extra patch. Thanks, Sean
[USD_Patch_for_Matt_2.patch.txt](https://github.com/Autodesk/maya-usd/files/4489164/USD_Patch_for_Matt_2.patch.txt)

**Comment by mattyjams** [comment]
> @mattyjams Sorry we found a problem when building on the Mac. Could you add in this one extra patch. Thanks, Sean
> [USD_Patch_for_Matt_2.patch.txt](https://github.com/Autodesk/maya-usd/files/4489164/USD_Patch_for_Matt_2.patch.txt)

No problem. Just pushed that as commit 4eb3d237b700e01857eff50697949b4683f0cf9a.

**Comment by mattyjams** [comment]
@kxl-adsk: Ahh, sorry about that. Should be fixed by commit 2ab470aa05adaa2e098aee37065d0afd4ac23112.

**Pull request merged**